UnityProject/Assets/GunPKA.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class GunPKA : Gun
+{
+
+	bool allowRecharge = true;
+	public float rechargeTime = 2.0f;
+	public override void ServerPerformInteraction(AimApply interaction)
+	{
+		ServerShoot(interaction.Performer, interaction.TargetVector.normalized, UIManager.DamageZone, false);
+		if (allowRecharge)
+		{
+			StartCoroutine(StartCooldown());
+		}
+	}
+	private IEnumerator StartCooldown()
+	{
+		allowRecharge = false;
+		yield return WaitFor.Seconds(rechargeTime);
+		CurrentMagazine.ExpendAmmo(-1);
+		SoundManager.PlayNetworkedAtPos("ReloadKinetic", gameObject.AssumedWorldPosServer());
+		allowRecharge = true;
+	}
+}
```

Comment:
I think some of this cooldown logic could be useful for future guns so it might be worth adding to Gun.cs at some point. This is fine for now though.